File: designpatterns/src/com/renho/behavior/ChainofResponsibility/ConcreteHandlerB.java
package com.renho.behavior.ChainofResponsibility;

public class ConcreteHandlerB extends Handler {

	@Override
	public void handleRequest(int request) {
		System.out.println("ConcreteHandlerB:" + request);
		if(request > 2) {
			System.out.println("无法处理,向上请求!");
			this.successor.handleRequest(request);			
		}else {
			System.out.println("ConcreteHandlerB:结束");
		}
	}

}

File: ideaworkspace/designpatterns/src/main/scala/com/renho/behavior/chainofresponsibility/ConcreteHandlerB.scala
package com.renho.behavior.chainofresponsibility

/**
  * Created by renho on 2016/7/11.
  */
class ConcreteHandlerB extends Handler {
	override def handlerRequest(request: Int): Unit = {
		if(request > 2) {
			println("无法处理, 向上请求!")
			successor.handlerRequest(request)
		} else {
			println("ConcreteHandlerB结束!")
		}
	}
}
